### PR TITLE
updated to support Koa >2

### PR DESCRIPTION
**Pull request by davidlondono**

### CODE REVIEW

**Comment by KingWu** [comment]
Wait for this pull

**Comment by dlau** [comment]
This is great!

I'll take a look tonight! There also seems to be some test failures as well as an `it.only` committed at first glance.

**Comment by dlau** [comment]
Merged into https://github.com/dlau/koa-body/tree/koa2 branch.

`npm install koa-body@2`

File: index.js [hunk 1]
```diff
@@ -33,52 +33,68 @@ function requestbody(opts) {
...
-        }
-        else if (opts.multipart && this.is('multipart')) {
-          body = yield formy(this, opts.formidable);
-        }
-        
-      } catch(parsingError) {
-        if (typeof(opts.onError) === 'function') {
-          opts.onError(parsingError, this);
+      } catch (parsingError) {
+        if (typeof opts.onError === 'function') {
```
Comment (tunnckoCore) [inline index.js]: never comes here

File: index.js [hunk 2]
```diff
@@ -33,52 +33,68 @@ function requestbody(opts) {
...
   opts.strict = 'strict' in opts ? opts.strict : true;
 
-  return function *(next){
-    var body = {};
+  return function (ctx, next) {
+    var bodyPromise;
     // so don't parse the body in strict mode
-    if (!opts.strict || ["GET", "HEAD", "DELETE"].indexOf(this.method.toUpperCase()) === -1) {
+    if (!opts.strict || ["GET", "HEAD", "DELETE"].indexOf(ctx.method.toUpperCase()) === -1) {
       try {
```
Comment (tunnckoCore) [inline index.js]: never throws

Comment (tunnckoCore) [inline index.js]: completely useless `try/catch` block

File: index.js [hunk 3]
```diff
@@ -91,42 +107,41 @@ function requestbody(opts) {
...
-      .on('field', function(field, value) {
-        if (fields[field]) {
-          if (Array.isArray(fields[field])) {
-            fields[field].push(value);
-          } else {
-            fields[field] = [fields[field], value];
-          }
+    var form = new forms.IncomingForm(opts);
+    form.on('end', function () {
+      return resolve({
```
Comment (tunnckoCore) [inline index.js]: no need to `return`

File: index.js [hunk 4]
```diff
@@ -91,42 +107,41 @@ function requestbody(opts) {
...
+        fields: fields,
+        files: files
+      });
+    }).on('error', function (err) {
+      return reject(err);
```
Comment (tunnckoCore) [inline index.js]: `.on('error', reject)` is enough

**Pull request closed without merging**